pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -380,7 +404,7 @@ func (r *ReconcileWindowsMachine) Reconcile(request reconcile.Request) (reconcil
 }
 
 // addWorkerNode configures the given Windows VM, adding it as a node object to the cluster
-func (r *ReconcileWindowsMachine) addWorkerNode(ipAddress, providerName, instanceID string) error {
+func (r *ReconcileWindowsMachine) addWorkerNode(ipAddress, providerName, instanceID, hostname string) error {
```

Comment:
This is going to conflict with the vSphere changes I think https://github.com/openshift/windows-machine-config-operator/pull/149/commits/410ad4c900fe4bdc383723f567bb21a052fb7227

Reply:
I suggest we normalize around requiring just a single input if possible.

Reply:
Yes, would need some changes. Does the changed hostname in vsphere matches the Node name ?

Reply:
It ensures hostname of the Windows VM matches the machine name, only for vSphere

Reply:
As long as the passed value matches the targeted node name, we should be good